### PR TITLE
My Account - Create wishlist UI

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

Added the Create Wishlist logic.

Note: This feature is built ahead of the GQL support. Due to that, the create list mutation is not available and will be handled in PWA-989.

Note to @schensley: The mobile mocks had the buttons placed below the radio button group but since I am using the dialog component they are placed at the bottom. Do we have a standard that we would like to follow for all the mobile dialogs/drawers about where to place the buttons?

## Related Issue

Closes [PWA-629](https://jira.corp.magento.com/browse/PWA-629)

### Verification Stakeholders

@schensley 

### Verification Steps
1. On the deployed app go to `/wishlist`.
2. Click on the `Create a List` button.
3. Fill out the dialog.

## Screenshots / Screen Captures (if appropriate)

image

image

## Checklist

- [ ] I have added tests to cover my changes, if necessary.
- [ ] I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Comment by schensley** [comment]
@revanth0212 this work looks really good.  UX approved.

File: packages/peregrine/lib/talons/WishlistPage/useCreateWishlist.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { useState, useCallback } from 'react';
+
+/**
+ * @function
```
Comment (larsroettig) [inline packages/peregrine/lib/talons/WishlistPage/useCreateWishlist.js:4]: Why @function js doc ?

File: packages/venia-ui/lib/components/WishlistPage/createWishlist.css [hunk 2]
```diff
@@ -1,3 +1,52 @@
+.root {
```
Comment (larsroettig) [inline packages/venia-ui/lib/components/WishlistPage/createWishlist.css:1]: Empty root style ?

File: packages/venia-ui/lib/components/WishlistPage/createWishlist.css [hunk 3]
```diff
@@ -11,6 +60,15 @@
...
+    grid-row: 1 / span 1;
+    grid-column: 1 / span 2;
+}
+
+.radioRoot {
+    display: grid;
+    gap: 1rem;
+
+    grid-row: 2 / span 1;
+    grid-column: 1 / span 2;
```
Comment (tjwiebell) [inline packages/venia-ui/lib/components/WishlistPage/createWishlist.css:73]: Shouldn't need any of this explicit positioning, the grid would flow this way naturally.

```suggestion
.radioRoot {
    display: grid;
    gap: 1rem;
```

File: packages/venia-ui/lib/components/WishlistPage/createWishlist.css [hunk 4]
```diff
@@ -1,3 +1,52 @@
+.root {
+}
+
+.body {
+    display: block;
```
Comment (tjwiebell) [inline packages/venia-ui/lib/components/WishlistPage/createWishlist.css:5]: I think this is attached to a `div`, shouldn't be needed.
```suggestion
```

File: packages/venia-ui/lib/components/WishlistPage/createWishlist.css [hunk 5]
```diff
@@ -1,3 +1,52 @@
...
+    overflow: auto;
+}
+
+.buttons {
+    padding: 1rem;
+    padding-bottom: 2.5rem;
```
Comment (tjwiebell) [inline packages/venia-ui/lib/components/WishlistPage/createWishlist.css:11]:
```suggestion
    padding: 1rem 1rem 2.5rem;
```

File: packages/venia-ui/lib/components/WishlistPage/__tests__/createWishlist.ee.spec.js [hunk 6]
```diff
@@ -4,8 +4,22 @@ import { createTestInstance } from '@magento/peregrine';
...
 
 jest.mock('../../../classify');
+jest.mock('../../Dialog', () => props => (
+    <div componentName={'Dialog'} {...props} />
+));
+jest.mock(
+    '@magento/peregrine/lib/talons/WishlistPage/useCreateWishlist',
+    () => ({
+        useCreateWishlist: jest.fn().mockReturnValue({
+            createList: jest.fn(),
```
Comment (tjwiebell) [inline packages/venia-ui/lib/components/WishlistPage/__tests__/createWishlist.ee.spec.js:14]: I've found it beneficial to name these mock functions, easier to spot breakages in the snaps.

```suggestion
            createList: jest.fn().mockName('createList'),
```

**Review by tjwiebell — CHANGES_REQUESTED** [review]
Looks good, some suggestions to clean things up.

File: packages/venia-ui/lib/components/WishlistPage/createWishlist.ee.js [hunk 7]
```diff
@@ -16,19 +31,109 @@ const CreateWishlist = props => {
...
+                    >
+                        <TextInput
+                            field="listname"
+                            validate={isRequired}
+                            validateOnBlur
+                        />
+                    </Field>
+                    <RadioGroup
+                        classes={radioGroupClasses}
+                        field="listtype"
```
[the remark this answers is not in the document]
Comment (revanth0212) [reply packages/venia-ui/lib/components/WishlistPage/createWishlist.ee.js:100]: I feel that looks like a mistake. `sharing_code` seems to be part of the `WishList` interface though, hence I would say, let's have it in the UX till the `WishList` feature is available through GQL.

File: packages/peregrine/lib/talons/WishlistPage/useCreateWishlist.js (hunk 1 above)
Comment (revanth0212) [reply packages/peregrine/lib/talons/WishlistPage/useCreateWishlist.js:4]: This is part of the new Docs initiate we have started for the code. I modeled this after @jcalcaben's previous PR. @jcalcaben is this as expected?

Comment (jcalcaben) [reply packages/peregrine/lib/talons/WishlistPage/useCreateWishlist.js:4]: @larsroettig this is used to help me auto-generate reference docs from the source code

@revanth0212 yes, this looks like the proper format

**Review by tjwiebell — APPROVED** [review]
Please loop in the GraphQL team just in case we need to add list visibility as an input field; this looks perfect with current coverage though. Nice work 👍

**Comment by revanth0212** [comment]
> Please loop in the GraphQL team just in case we need to add list visibility as an input field; this looks perfect with current coverage though. Nice work 👍

Thanks, @tjwiebell. I have created an [issue](https://github.com/magento/architecture/issues/453) on the architecture board asking this question.

**Comment by dpatil-magento** [comment]
@revanth0212 Please check below observations - 
1. Per mocks `Private` radio should be first and `Public` below it, but app shows opposite, expected?
2. The entire Row is clickable,  I think only the link should be clickable?

<a href="https://gyazo.com/a299661fd293bec304cf07315cfd9b94">Image from Gyazo</a>

**Comment by revanth0212** [comment]
> @revanth0212 Please check below observations -
> 
> 1. Per mocks `Private` radio should be first and `Public` below it, but app shows opposite, expected?
> 2. The entire Row is clickable,  I think only the link should be clickable?
> 
> [Image from Gyazo](https://gyazo.com/a299661fd293bec304cf07315cfd9b94)

Thanks for the check Dev. I have addressed the order of the radio buttons. Regarding the link, I haven't changed it and I think it is as expected.

**Comment by schensley** [comment]
@dpatil-magento @revanth0212 
Regarding your question (2. above), ideally the entire area (row, button, container) should be clickable as it behaves in the demo. Provides for a much bigger target for the shopper.

**Review by tjwiebell — APPROVED** [review]

**Pull request merged**